### DISCUSSION TITLE
I am curious about the role of informer in each module of Kubernetes scheduler.

### DISCUSSION THREAD

**Discussion by kschoi93** [kubernetes-sigs/scheduler-plugins#705] [Q&A] [answered]
hello? I am studying kubernetes scheduler.

I would like to know the difference between the informer in the controller and the informer in the scheduler.
It appears that the controller's informer only performs add, modify, and delete operations without any special action.
The scheduler's informer identifies the execution of a specific task, but the controller's informer is not understood.

And what exactly is the role of the controller in the Kubernetes scheduler plugins?

* The version I checked while developing was version 1.24.

The Kubernetes scheduler feels big to me, making it difficult to analyze. 
I hope you can help me. thank you

**Comment by Huang-Wei** [comment, marked as answer]
I assume by "controller", you meant the controllers placed in https://github.com/kubernetes-sigs/scheduler-plugins/tree/master/pkg/controllers. If so, those controllers are merely reflecting the latest status of the CR objects that some scheduler plugin interacts with. The controller should be only updating the `.status` portion of the CR objects, and it should be independent with scheduler plugins' logic and scheduling decision.

In short, you can treat "controllers" (in the context of scheduler plugins) as facilitating components for better user experience. And scheduler plugin can live w/o the controllers.

**Reply by kschoi93** [reply]
Your answer was perfect. 

That was the answer I wanted. 

Thanks to this, my head felt at ease.
